core/src/main/java/org/mule/runtime/core/api/transformer/Transformer.java

```diff
@@ -95,4 +97,14 @@
    * @since 3.0.0
    */
   DataType getReturnDataType();
+
+  @Override
+  default ProcessingType getProccesingType() {
+    if (getSourceDataTypes().stream().filter(dataType -> !dataType.isStreamType()).count() > 0) {
+      return IO_RW;
+    } else {
+      return CPU_LITE;
```

Comment:
Only complex transformers should be CPU intensive, most of the built-in transformers are very simple.  I can make some changes to improve this though..